piv/key_test.go

```diff
@@ -683,3 +683,166 @@ func TestRetiredKeyManagementSlot(t *testing.T) {
 		})
 	}
 }
+
+func TestSetRSAPrivateKey(t *testing.T) {
+	tests := []struct {
+		name    string
+		bits    int
+		slot    Slot
+		wantErr error
+	}{
+
+		{
+			name:    "rsa 1024",
+			bits:    1024,
+			slot:    SlotSignature,
+			wantErr: nil,
+		},
+		{
+			name:    "rsa 2048",
+			bits:    2048,
+			slot:    SlotCardAuthentication,
+			wantErr: nil,
+		},
+		{
+			name:    "rsa 4096",
+			bits:    4096,
+			slot:    SlotAuthentication,
+			wantErr: errUnsupportedKeySize,
+		},
+		{
+			name:    "rsa 512",
+			bits:    512,
+			slot:    SlotKeyManagement,
+			wantErr: errUnsupportedKeySize,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			yk, close := newTestYubiKey(t)
+			defer close()
+
+			generated, err := rsa.GenerateKey(rand.Reader, tt.bits)
+			if err != nil {
+				t.Fatalf("generating private key: %v", err)
+			}
+
+			err = yk.SetPrivateKeyInsecure(DefaultManagementKey, tt.slot, generated, Key{
+				PINPolicy:   PINPolicyNever,
+				TouchPolicy: TouchPolicyNever,
+			})
+			if err != nil {
```

Comment:
It looks like you're missing if you didn't get an error but wanted one. Suggest changing this to

```go
if (err != nil) != tt.wantErr {
    t.Fatalf("SetPrivateKeyInsecure(): wantErr=%v, got err=%v", tt.wantErr, err)
}
if err != nil {
    return
}
```